packages/ethers/src/client.ts

```diff
@@ -912,7 +912,17 @@ export class Web3Modal extends Web3ModalScaffold {
             ? accounts
             : [{ address, type: preferredAccountType as 'eoa' | 'smartAccount' }]
         )
-        EthersStoreUtil.setChainId(chainId)
+
+        /**
+         * Warning: This is a temporary fix to handle both string and number chainId
+         * `chainId` will be only string after the dependencies are updated
+         */
+        if (typeof chainId === 'string') {
+          EthersStoreUtil.setChainId(NetworkUtil.caipNetworkIdToNumber(chainId as CaipNetworkId))
+        } else {
+          EthersStoreUtil.setChainId(chainId)
```

Comment:
this is repeated in wagmi and here => I'd abstreact it to `@web3modal/common/NetworkUtil`